exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -19,14 +18,15 @@ import (
 	elasticsearch7 "github.com/elastic/go-elasticsearch/v7"
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
+	"golang.org/x/sync/semaphore"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/sanitize"
 )
 
 type esClientCurrent = elasticsearch7.Client
 type esConfigCurrent = elasticsearch7.Config
 
-type esBulkIndexerCurrent = bulkIndexerPool
+type esBulkIndexerCurrent = bulkIndexerManager
 
 type esBulkIndexerItem = docappender.BulkIndexerItem
```

Comment:
I don't get what's the idea behind giving new names to types:
- `esBulkIndexerCurrent = bulkIndexerManager`
- `esBulkIndexerItem = docappender.BulkIndexerItem`

If it's to abstract docapender implementation details, I don't think it does the job, as `worker.addBatchAndFlush` requires `esBulkIndexerItem` to be a `docappender.BulkIndexerItem` anyway.

I think this file could use a bit of refactoring to make it easier to comprehend (non-blocking, this is an idea for another PR). Or maybe it's just me reading this code for the first time. :monocle_face:

Reply:
I believe it was there to simplify the switch between v7 and v8 go-elasticsearch, and the argument no longer stands as we moved to docappender. I've removed the type aliases.